src/platform_impl/web/event_loop/window_target.rs

```diff
@@ -558,6 +525,33 @@ impl<T> EventLoopWindowTarget<T> {
                 event: WindowEvent::ThemeChanged(theme),
             });
         });
+
+        canvas.on_resize_scale(
+            {
+                let runner = self.runner.clone();
+                let canvas = canvas_clone.clone();
+
+                move |size, scale| {
+                    runner.send_event(EventWrapper::ScaleChange {
+                        canvas: Rc::downgrade(&canvas),
+                        size,
+                        scale,
+                    })
+                }
+            },
+            {
+                let runner = self.runner.clone();
+
+                move |size| {
+                    RefCell::borrow(&canvas_clone).set_inner_size(size);
+                    runner.send_event(Event::WindowEvent {
+                        window_id: RootWindowId(id),
+                        event: WindowEvent::Resized(size),
```

Comment:
Apparently changing the writing mode can cause the `ResizeObserver` to fire if `devicePixelContentBoxSize` is used even if the size hasn't changed.
So we actually have to check if the size changed before firing the event, which means we have to be careful about when we actually use `set_inner_size()`, otherwise we can't do the check properly.

Reply:
Done in https://github.com/rust-windowing/winit/pull/2859/commits/0eb1d3ece3f2b4fad8f0cb5285c377e433f08712.